### PR TITLE
Accessibility: Screen reader accessible help button on forms

### PR DESCRIPTION
**Pull request by pxwxnvermx**
## Product Description
This PR adds screen reader fixes for the help button for questions on a form.

## Technical Summary
[Ticket Link](https://dimagi-dev.atlassian.net/browse/USH-1989)

### Safety story
Tested locally with Voiceover screen reader.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by mkangia — APPROVED** [review]
Good to run it by USH

**Comment by stephherbers** [comment]
Looks good but are we thinking QA?

**Comment by esoergel** [comment]
(@pxwxnvermx FYI I've asked @stephherbers to review for USH as she has the most prior experience with accessibility standards)

File: corehq/apps/cloudcare/templates/form_entry/templates.html [hunk 1]
```diff
@@ -202,11 +202,17 @@ <h1 class="title" data-bind="text: title, visible: !showInFormNavigation()"></h1
...
+        <span class="webapp-markdown-output" tabindex="0"
+              data-bind="
+                html: ko.utils.unwrapObservable($data.caption_markdown) || caption(),
+                css: {'col-md-11' : help()}"></span>
         <!-- ko if: help() -->
         <a
-          class="help-text-trigger"
+          class="help-text-trigger col-md-1"
           role="button"
+          href="javascript:void(0)"
```
Comment (stephherbers) [inline corehq/apps/cloudcare/templates/form_entry/templates.html:214]: this line just shows the dialog without refreshing the page, right? Same this as if we said something like `onclick="return false";` ?

**Review by stephherbers — APPROVED** [review]
looking good we'll see what QA says!

File: corehq/apps/cloudcare/templates/form_entry/templates.html (hunk 1 above)
Comment (pxwxnvermx) [reply corehq/apps/cloudcare/templates/form_entry/templates.html:214]: Yep, without the href attribute this link was not getting focused on <tab> key, so using `href="javascript:void(0)"` it gets focused and causes no redirect/refresh on click.

**Comment by pxwxnvermx** [comment]
@stephherbers The QA for this ticket has been done and details regarding the QA are updated on the Ticket itself.

**Pull request merged**